src/components/ImageUpload.tsx

```diff
@@ -1,66 +1,130 @@
-import { Paper, Typography } from "@mui/material";
-import { styled } from "@mui/material/styles";
-import { Box } from "@mui/system";
-import { Field, FormikErrors, FormikTouched } from "formik";
-import React from "react";
-import { render } from "react-dom";
+import { Box, CircularProgress, Typography } from "@mui/material";
+import { Image } from "cloudinary-react";
+import { useCallback, useState } from "react";
 import { useDropzone } from "react-dropzone";
-import { theme } from "./../theme/theme";
-import { StyledPaperInput } from "./StyledComponents";
+import { StyledFilledButton, StyledPaperInput } from "./StyledComponents";
+
+// once we have backend authentication we will need to protect these variables
+const PUBLIC_CLOUDINARY_CLOUD_NAME = "projectnewm",
+  PUBLIC_CLOUDINARY_UPLOAD_PRESET = "rvktckuk";
 
 interface ImageUploadProps {
-  setFieldValue: (field: string, value: any, shouldValidate?: boolean | undefined) => void;
-  errors: FormikErrors<{
-    description: string;
-    genre: string;
-    imageFile: {
-      name: string;
-      size: string;
-      type: string;
-    };
-  }>;
-  touched: FormikTouched<{
-    description: string;
-    genre: string;
-    imageFile: {
-      name: string;
-      size: null;
-      type: string;
-    };
-    releaseDate: string;
-    title: string;
-    yourRole: string;
-  }>;
+  fileType?: "png" | "jpg" | "jpeg" | "wdp" | "jp2" | "bmp" | "pdf" | "tiff" | "ico" | "eps";
+  dropzoneLabel?: string;
+  buttonLabel?: string;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  setFieldValue?: (field: string, value: any, shouldValidate?: boolean | undefined) => void;
+  setTouched?: (fields: { [field: string]: boolean }, shouldValidate?: boolean) => void;
 }
 
-export const ImageUpload = (props: ImageUploadProps) => {
-  const { setFieldValue } = props;
+export const ImageUpload = ({
+  fileType = "png",
+  dropzoneLabel = "Drag & Drop (Square Image Only)",
+  buttonLabel = "Upload Image",
+  setFieldValue,
+  setTouched,
+}: ImageUploadProps) => {
+  const [imgLoaded, setImgLoaded] = useState<boolean>(false);
+  const [uploadedFilePublicId, setUploadedFilePublicId] = useState<string>();
+  const onDrop = useCallback(
+    acceptedFiles => {
+      const url = `https://api.cloudinary.com/v1_1/${PUBLIC_CLOUDINARY_CLOUD_NAME}/upload`;
+
+      acceptedFiles.forEach(async (acceptedFile: File) => {
+        const formData = new FormData();
+        formData.append("file", acceptedFile);
+        formData.append("upload_preset", String(PUBLIC_CLOUDINARY_UPLOAD_PRESET));
+
+        const response = await fetch(url, {
+          body: formData,
+          method: "post",
+        });
+
+        const data = await response.json();
+        setUploadedFilePublicId(data.public_id);
```

Comment:
not for this pr, but we should add an error handler here.  If the call to the cloudinary api fails, we'd probably want to let the user know so they can retry

Reply:
Created a click up task for this